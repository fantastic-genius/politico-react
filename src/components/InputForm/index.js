import React from 'react';
import PropTypes from 'prop-types';

const InputForm = ({ labelName, type, id, name, placeholder, ...rest }) => (
  <div>
    <label htmlFor={ name }>{ labelName }</label>
    <input type={ type } id={ id } name={ name } placeholder={ placeholder } {...rest}/>
  </div>
);

InputForm.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  labelName: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool
}

InputForm.defaultTypes = {
  id: null,
  type:'text',
  labelName: null,
  name: null,
  placeholder: null,
  required: false
}

export default InputForm;
