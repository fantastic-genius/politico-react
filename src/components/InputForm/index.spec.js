import React from 'react';
import { shallow } from 'enzyme';
import InputForm from './index';

describe('<InputForm />', () => {
  describe('Rendering', () => {
    it('should render Footer correctly in debug mode', () => {
        const component = shallow(<InputForm debug/>);
        expect(component).toMatchSnapshot();
    });
  })

});
