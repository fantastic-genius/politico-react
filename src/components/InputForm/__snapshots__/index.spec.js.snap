// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm /> Rendering should render Footer correctly in debug mode 1`] = `
<div>
  <label />
  <input
    debug={true}
  />
</div>
`;
